src/ore/src/assert.rs

```diff
@@ -14,6 +14,62 @@
 // limitations under the License.
 
 //! Assertion utilities.
+//!
+//! # Optional assertions
+//!
+//! Optional assertions are like debug assertions, but they can be toggled on
+//! and off at runtime in a release build.
```

Comment:
Nit: I think we can only enable/disable the assertions at startup time, but once they are set, they shouldn't be changed anymore. So it's not exactly "can be toggled [...] at runtime".